OpenEdX/Managers/PushNotificationsManager/PushNotificationsManager.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  PushNotificationManager.swift
+//  OpenEdX
+//
+//  Created by Anton Yarmolenka on 24/01/2024.
+//
+
+import Foundation
+import Core
+import UIKit
+import Swinject
+
+public protocol PushNotificationsProvider {
+    func registerWithDeviceToken(deviceToken: Data)
+    func didFailToRegisterForRemoteNotificationsWithError(error: Error)
+}
+
+protocol PushNotificationsListener {
+    func shouldListenNotification(userinfo: [AnyHashable: Any]) -> Bool
```

Comment:
Is this method necessary? as if a push listener is register it should be listening to the push notifications. Or there will be other cases?

Reply:
Braze and FCM notifications have different formats, and if we have two listeners enabled, when calling the `didReceiveRemoteNotification` function, we must determine that it is a notification from the correct service. Otherwise we will try to retrieve the data from the wrong format and it will fail 99% of the time.

Reply:
It seems this method is to check whether the notification is for the listener or not. If that so who about renaming it to like canProcessPushNotification or something like that thoughts?

Reply:
This method is not for checking `can or not` process this notification. It's for checking `should or not` this Listener handle with this notification at all (or this is for other Listener).
We're developing an abstraction layer, aren't? I believe we can rename any function during implementation if we see it does something other, no?